app/services/session_encryptor.rb

```diff
@@ -1,25 +1,27 @@
 class SessionEncryptor
-  def self.build_user_access_key
-    key = Figaro.env.session_encryption_key
-    UserAccessKey.new(password: key, salt: key)
+  def user_access_key
+    @user_access_key ||= begin
+      key = Figaro.env.session_encryption_key
+      uak = UserAccessKey.new(password: key, salt: key)
+      uak.random_r = OpenSSL::Digest::SHA256.digest(key)
+      uak
+    end
```

Comment:
Is there a reason we switched this to instance methods? If I'm reading this correctly, we will be generating a new UserAccessKey on every request, whereas before we weren't. I reverted to the class methods we had before, but added the`uak.random_r` from this PR, and the test you added passed.

Reply:
This makes it easier for us to test and reproduce the error

We still have one instance and one `user_access_key` for the lifetime of the app

Reply:
The `serializer: SessionEncryptor.new` is esentially the same was the old `serializer: SessionEncryptor` with the singleton class

Reply:
We aren't running this on every request. We changed the session store config [here](https://github.com/18F/identity-idp/pull/1718/files#diff-ae8f555b82c0416edbcf62c9c1a5b73bR17) to use a singleton session encryptor.

We made this into an object so we could test that 2 instances of the session encryptor will encrypt / decrypt with the same results.

Reply:
Scratch that. I don't think my test is valid because I'm calling `SessionEncryptor` twice. It's the same object ID.

Reply:
Gotcha. Thanks for the clarification.